Crashes reported by PlayStore (new in 1.0.14)

- [x] Number 1
```
java.lang.NullPointerException: 
  at im.vector.app.core.platform.VectorBaseFragment.getViews (VectorBaseFragment.kt:1)
  at im.vector.app.features.home.room.detail.RoomDetailFragment.access$getViews$p (RoomDetailFragment.kt:1)
  at im.vector.app.features.home.room.detail.RoomDetailFragment$updateJumpToReadMarkerViewVisibility$1$1.invoke (RoomDetailFragment.kt:11)
  at im.vector.app.features.home.room.detail.RoomDetailFragment$updateJumpToReadMarkerViewVisibility$1$1.invoke (RoomDetailFragment.kt:1)
  at androidx.preference.R$string.withState (Unknown Source:1)
  at im.vector.app.features.home.room.detail.RoomDetailFragment$updateJumpToReadMarkerViewVisibility$1.run (RoomDetailFragment.kt:1)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:223)
  at android.app.ActivityThread.main (ActivityThread.java:7660)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:592)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:947)
```

- [x] Number 2
To reproduce: Search something inside the timeline and hit back before the search results appears.
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performDestroyActivity (ActivityThread.java:5193)
  at android.app.ActivityThread.handleDestroyActivity (ActivityThread.java:5227)
  at android.app.servertransaction.DestroyActivityItem.execute (DestroyActivityItem.java:44)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2147)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:7814)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1068)
Caused by: java.lang.NullPointerException: 
  at im.vector.app.core.platform.VectorBaseFragment.getViews (VectorBaseFragment.kt:1)
  at im.vector.app.features.home.room.detail.search.SearchFragment.onDestroy (SearchFragment.kt:2)
  at androidx.fragment.app.Fragment.performDestroy (Fragment.java:6)
  at androidx.fragment.app.FragmentStateManager.destroy (FragmentStateManager.java:28)
  at androidx.fragment.app.FragmentStateManager.moveToExpectedState (FragmentStateManager.java:57)
  at androidx.fragment.app.SpecialEffectsController$FragmentStateManagerOperation.complete (SpecialEffectsController.java:2)
  at androidx.fragment.app.SpecialEffectsController$Operation.cancel (SpecialEffectsController.java:4)
  at androidx.fragment.app.SpecialEffectsController.forceCompleteAllOperations (SpecialEffectsController.java:14)
  at androidx.fragment.app.FragmentManager.dispatchStateChange (FragmentManager.java:8)
  at androidx.fragment.app.FragmentManager.dispatchDestroy (FragmentManager.java:4)
  at androidx.fragment.app.FragmentActivity.onDestroy (FragmentActivity.java:3)
  at androidx.appcompat.app.AppCompatActivity.onDestroy (AppCompatActivity.java:1)
  at im.vector.app.core.platform.VectorBaseActivity.onDestroy (VectorBaseActivity.kt:1)
  at android.app.Activity.performDestroy (Activity.java:8217)
  at android.app.Instrumentation.callActivityOnDestroy (Instrumentation.java:1342)
  at android.app.ActivityThread.performDestroyActivity (ActivityThread.java:5178)
```
- [x] Number 3
To reproduce: Open emoji keyboard. Hit back from the navigation bar.
```
java.lang.NullPointerException: 
  at im.vector.app.core.platform.VectorBaseFragment.getViews (VectorBaseFragment.kt:1)
  at im.vector.app.features.home.room.detail.RoomDetailFragment.access$getViews$p (RoomDetailFragment.kt:1)
  at im.vector.app.features.home.room.detail.RoomDetailFragment$setupEmojiPopup$emojiPopup$2.onEmojiPopupDismiss (RoomDetailFragment.kt:1)
  at com.vanniktech.emoji.EmojiPopup$6.onDismiss (EmojiPopup.java:4)
  at android.widget.PopupWindow.dismiss (PopupWindow.java:1937)
  at com.vanniktech.emoji.EmojiPopup.dismiss (EmojiPopup.java:1)
  at com.vanniktech.emoji.EmojiPopup$2.onViewDetachedFromWindow (EmojiPopup.java:2)
  at android.view.View.dispatchDetachedFromWindow (View.java:18436)
  at android.view.ViewGroup.dispatchDetachedFromWindow (ViewGroup.java:3777)
  at android.view.ViewRootImpl.dispatchDetachedFromWindow (ViewRootImpl.java:3831)
  at android.view.ViewRootImpl.doDie (ViewRootImpl.java:6783)
  at android.view.ViewRootImpl.die (ViewRootImpl.java:6760)
  at android.view.WindowManagerGlobal.removeViewLocked (WindowManagerGlobal.java:458)
  at android.view.WindowManagerGlobal.removeView (WindowManagerGlobal.java:396)
  at android.view.WindowManagerImpl.removeViewImmediate (WindowManagerImpl.java:126)
  at android.app.ActivityThread.handleDestroyActivity (ActivityThread.java:4520)
  at android.app.servertransaction.DestroyActivityItem.execute (DestroyActivityItem.java:39)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:145)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1814)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:280)
  at android.app.ActivityThread.main (ActivityThread.java:6706)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,7 @@ Improvements 🙌:
 Bugfix 🐛:
  - Fix clear cache issue: sometimes, after a clear cache, there is still a token, so the init sync service is not started.
  - Sidebar too large in horizontal orientation or tablets (#475)
+ - Crashes reported by PlayStore (new in 1.0.14) (#2707)
 
 Translations 🗣:
  -

--- a/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt
+++ b/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt
@@ -196,6 +196,8 @@ class UiAllScreensSanityTest {
         pressBack()
         clickMenu(R.id.video_call)
         pressBack()
+        clickMenu(R.id.search)
+        pressBack()
 
         pressBack()
     }

--- a/vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailFragment.kt
+++ b/vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailFragment.kt
@@ -297,6 +297,8 @@ class RoomDetailFragment @Inject constructor(
     private var lockSendButton = false
     private val activeCallViewHolder = ActiveCallViewHolder()
 
+    private lateinit var emojiPopup: EmojiPopup
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         sharedActionViewModel = activityViewModelProvider.get(MessageSharedActionViewModel::class.java)
@@ -512,7 +514,7 @@ class RoomDetailFragment @Inject constructor(
     }
 
     private fun setupEmojiPopup() {
-        val emojiPopup = EmojiPopup
+        emojiPopup = EmojiPopup
                 .Builder
                 .fromRootView(views.rootConstraintLayout)
                 .setKeyboardAnimationStyle(R.style.emoji_fade_animation_style)
@@ -591,6 +593,7 @@ class RoomDetailFragment @Inject constructor(
         autoCompleter.clear()
         debouncer.cancelAll()
         views.timelineRecyclerView.cleanup()
+        emojiPopup.dismiss()
 
         super.onDestroyView()
     }

--- a/vector/src/main/java/im/vector/app/features/home/room/detail/search/SearchFragment.kt
+++ b/vector/src/main/java/im/vector/app/features/home/room/detail/search/SearchFragment.kt
@@ -76,10 +76,10 @@ class SearchFragment @Inject constructor(
         controller.listener = this
     }
 
-    override fun onDestroy() {
-        super.onDestroy()
+    override fun onDestroyView() {
         views.searchResultRecycler.cleanup()
         controller.listener = null
+        super.onDestroyView()
     }
 
     override fun invalidate() = withState(searchViewModel) { state ->

